examples/nest/structural_plasticity.py

```diff
@@ -0,0 +1,234 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import nest
+import numpy
+import matplotlib.pyplot as pl
+
+class StructralPlasticityExample:
+    def __init__(self):
+        # General simulation parameters
+        # simulated time (ms)
+        self.t_sim = 250000.0
+        # simulation step (ms).
+        self.dt = 0.1                                    
+        self.number_excitatory_neurons = 800             
+        self.number_inhibitory_neurons = 200            
+        
+        # Structural_plasticity and synapse properties
+        self.update_interval = 1000                       
+        self.record_interval = 1000.0
+        # rate of background Poisson input
+        self.bg_rate = 10000.0       
+        self.neuron_model = 'iaf_psc_exp'               
+        
+        # Growth curves for synaptic elements of excitatory neurons
+        # Excitatory synaptic elements 
+        self.growth_curve_e_e = {
+            'growth_curve': "gaussian",
+            'growth_rate': 0.0001,
+            'continuous': False,
+            'eta': 0.0,
+            'eps': 0.05,
+        }
+        
+        # Inhibitory synaptic elements 
+        self.growth_curve_e_i = {
+            'growth_curve': "gaussian",
+            'growth_rate': 0.0001,
+            'continuous': False,
+            'eta': 0.0,
+            'eps': self.growth_curve_e_e['eps'],
+        }
+        
+        # Growth curves for synaptic elements of inhibitory neurons 
+        # Excitatory synaptic elements 
+        self.growth_curve_i_e = {
+            'growth_curve': "gaussian",
+            'growth_rate': 0.0004,
+            'continuous': False,
+            'eta': 0.0,
+            'eps': 0.2,
+        }
+        
+        # Inhibitory synaptic elements
+        self.growth_curve_i_i = {
+            'growth_curve': "gaussian",
+            'growth_rate': 0.0001,
+            'continuous': False,
+            'eta': 0.0,
+            'eps': self.growth_curve_i_e['eps']
+        }
+        
+        
+        # Neuron model 
+        self.model_params = {'tau_m': 10.0,   # membrane time constant (ms)
+            'tau_syn_ex': 0.5,          # excitatory synaptic time constant (ms)
+            'tau_syn_in': 0.5,          # inhibitory synaptic time constant (ms)
+            't_ref': 2.0,               # absolute refractory period (ms)
+            'E_L': -65.0,               # resting membrane potential (mV)
+            'V_th': -50.0,              # spike threshold (mV)
+            'C_m': 250.0,               # membrane capacitance (pF)
+            'V_reset': -65.0,           # reset potential (mV)
+        } 
+        
+        self.nodes = ()
+        self.nodes_i = ()
```

Comment:
For symmetry, I would call the excitatory nodes `self.nodes_e`. And instead of initializing with an empty tuple, I would use `None`.